Automattic/woocommerce-payments
# Add redirect to view Capital financing offers

Fixes #3655 

#### Changes proposed in this Pull Request

This PR prepares the changes from #3673 to be included on `develop`. This includes merging the original changes and adding them behind a feature flag.

**Question 1:** The Capital work won't be very extensive, so I decided to apply the feature flag directly where appropriate instead of pulling all changes to a specific class. Does that make sense, or should we extract Capital related code to a separate class before merging?

**Question 2:** Should we add a changelog and QA testing instructions in this case? We're shipping a feature to the client, but it won't be included/usable by default.

#### Testing instructions

_The redirect view has been tested as part of #3673, so a quick smoke test should suffice here._

* Before enabling the feature flag:
    * Navigate to `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview&wcpay-loan-offer-error=1` and assert that no error messages show up
    * Navigate to `/wp-admin/admin.php?wcpay-loan-offer` and assert you get a blank page
* Install 47-gh-Automattic/woocommerce-payments-dev-tools and enable Stripe Capital
    * Navigate to `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview&wcpay-loan-offer-error=1` and assert that the "There was a problem redirecting you to the loan offer..." error is displayed
    * Navigate to `/wp-admin/admin.php?wcpay-loan-offer` and assert that you're redirected to the loan details page (or, if using the platform account, to the overview page as the request will fail) 

-------------------


**Post merge**
